icu4c/source/test/intltest/numfmtst.cpp

```diff
@@ -2120,6 +2120,17 @@ void NumberFormatTest::TestCurrencyUnit(void){
     static const char XXX8[]  =  "XXX";
     static const UChar INV[]  = u"{$%";
     static const char INV8[]  =  "{$%";
+    static const UChar ZZZ[]  = u"zz";
```

Comment:
Would it be maybe worth adding a BAD3 test for the embedded nul (U_ILLEGAL_ARGUMENT_ERROR)?

Reply:
This test case already exercises that path:

```cpp
} else if (_isoCode[1] == 0 || _isoCode[2] == 0) {
```

Down below, for the 8-bit char string case, I run the constructor both with length 2 and with length 3 to exercise both branches of the 8-bit constructor.

Reply:
Ah I missed this: `CurrencyUnit zzz8b({ZZZ8, 3}, ec); `

Thanks Shane!